hibernate-core/src/main/java/org/hibernate/internal/StatelessSessionImpl.java

```diff
@@ -358,6 +360,56 @@ private Object createProxy(EntityKey entityKey) {
 		return proxy;
 	}
 
+	@Override
+	public void fetch(Object association) {
+		checkOpen();
+		PersistenceContext persistenceContext = getPersistenceContext();
+		if ( association instanceof HibernateProxy) {
+			LazyInitializer initializer =
+					((HibernateProxy) association).getHibernateLazyInitializer();
+			if ( initializer.isUninitialized() ) {
+				String entityName = initializer.getEntityName();
+				Object id = initializer.getIdentifier();
+				EntityPersister persister =
+						getFactory().getMetamodel().entityPersister(entityName);
+				initializer.setSession(this);
+				persistenceContext.beforeLoad();
+				try {
+					Object entity = initializer.getImplementation(); //forces the load to occur
+					if ( entity==null ) {
+						getFactory().getEntityNotFoundDelegate().handleEntityNotFound( entityName, id );
+					}
+					initializer.setImplementation( entity );
+					initializer.unsetSession();
+				}
+				finally {
+					persistenceContext.afterLoad();
+					if ( persistenceContext.isLoadFinished() ) {
+						persistenceContext.clear();
+					}
+				}
+			}
+		}
+		else if ( association instanceof PersistentCollection) {
+			PersistentCollection<?> persistentCollection = (PersistentCollection<?>) association;
+			if ( !persistentCollection.wasInitialized() ) {
+				CollectionPersister persister =
+						getFactory().getMetamodel().collectionPersister( persistentCollection.getRole() );
+				Object key = persistentCollection.getKey();
+				persistenceContext.addUninitializedCollection( persister, persistentCollection, key );
+				persistentCollection.setCurrentSession(this);
```

Comment:
Shouldn't we also unset the session here in the finally block?

Reply:
Not as far as I am aware: I don't think SS generally cleans up it's temporary PC in a finally. (Remember: failures are unrecoverable in Hibernate, and the session is considered hosed after a failure.)

Reply:
Dunno, it just felt odd that you unset it in the above if-branch but not here. If you say it's fine, I'm ok with that. Just wanted to bring your attention to this.

Reply:
Oh you're saying I forget to unset it, not that you want me to unset it in a `finally`. Yeah, sure, I guess that would be more consistent.